ir/instr.cpp

```diff
@@ -1949,11 +1952,17 @@ StateValue ICmp::toSMT(State &s) const {
     fn = [this, &s, fn](const expr &av, const expr &bv, Cond cond) {
       Pointer lhs(s.getMemory(), av);
       Pointer rhs(s.getMemory(), bv);
-      if (use_provenance) {
+      switch (pcmode) {
+      case INTEGRAL:
+        return fn(lhs.getAddress(), rhs.getAddress(), cond);
+      case PROVENANCE:
         assert(cond == EQ || cond == NE);
         return cond == EQ ? lhs == rhs : lhs != rhs;
+      case OFFSETONLY:
```

Comment:
uhm, is this any useful?
It may well end up with different exprs in src/tgt because of syntactic differences in the IR. If something can be folded, the smt exprs will do it.

Reply:
I extracted offset because the meaning of ule/ult/... bids isn't clear. transform.cpp creates this mode when they point to the same base object, but without the knowledge comparing bids looks a bit counterintuitive.
I thought in terms complexity of expression it was equivalent. In which case does it cause syntactic difference?

Reply:
Well, comparing offsets only would be not very intuitive too. What I thought was to leave a comment at `OFFSETONLY` saying that this is meaningful only if the two pointers have same objects, but leaving it at `USEPROVENANCE` would work too.
In terms of syntactic difference I didn't understand yet.

Reply:
this is ok, i hadn't read the whole patch.
Essentially you wanted to avoid reinstating the old Pointer comparison operators. This is fine as long as we don't compare ptrs of different objs, which I guess it should be the common case.

Reply:
yep, the old pointer comparison operators are not necessary yet I think.